kubernetes-discovery-client/src/main/java/io/micronaut/kubernetes/configuration/KubernetesConfigMapWatcher.java

```diff
@@ -80,7 +83,9 @@ public KubernetesConfigMapWatcher(Environment environment, KubernetesClient clie
     @Override
     public void onApplicationEvent(ServiceReadyEvent event) {
         long lastResourceVersion = computeLastResourceVersion();
-        String labelSelector = computeLabelSelector(configuration.getConfigMaps().getLabels());
+        Map<String, String> labels = new HashMap<>(configuration.getConfigMaps().getLabels());
```

Comment:
`configuration.getConfigMaps().getLabels()` already returns a `Map<String, String>`, no need for a `new HashMap<>`

Reply:
I do this, because i do a `putAll` in the next line. Without the `new HashMap<>` it would modify the content of the source. When i remove the  `new HashMap<>` some of the tests fail.